backend/src/Edna/Analysis/FourPL.hs

```diff
@@ -45,8 +60,49 @@ instance ToSchema Params4PL where
   declareNamedSchema Proxy =
     declareNamedSchema @Params4PLTuple Proxy
 
--- | This function performs actual analysis and finds the best parameters.
+type AnalysisResult = Either Text Params4PL
+--  deriving stock (Generic, Show, Eq)
+--
+--deriveJSON ednaAesonWebOptions ''AnalysisResult
+--
+--instance Buildable AnalysisResult where
+--  build = buildFromJSON
 --
--- TODO [EDNA-71] Implement!
-analyse4PL :: [(Double, Double)] -> IO Params4PL
-analyse4PL _ = pure (Params4PL 1 2 3 4)
+--instance ToSchema AnalysisResult where
+--  declareNamedSchema Proxy =
+--    declareNamedSchema @AnalysisResult Proxy
+
+data Params4PLReq = Params4PLReq
+  { plreqExperiment :: ExperimentId
+  , plreqData :: [(Double, Double)]
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLReq
+
+data Params4PLResp = Params4PLResp
+  { plrspExperiment :: ExperimentId
+  , plrspData :: AnalysisResult
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLResp
+
+-- | This function performs actual analysis and finds the best parameters.
+analyse4PL :: [(ExperimentId, [Measurement])] -> Edna [Params4PLResp]
+analyse4PL experiments = do
+  let request = unpack $ encode $ flip map experiments $
+        \(eId, ms) -> Params4PLReq eId $ map (\m -> (mConcentration m, mSignal m)) ms
+  logMessage $ T.pack $ "Python data request: " <> request
```

Comment:
1. AFAIU the request may be quite big, so I think `logDebug` is more appropriate (I think you rarely need to see full requests, only when something goes wrong in analysis).
2. I prefer `toText`/`toString` functions, but that's up to you. I just don't remember in which direction `pack` converts and in which `unpack`.